app/code/core/Mage/Tax/Model/Resource/Calculation.php

```diff
@@ -222,16 +222,16 @@ protected function _collectPercent($percent, $rate)
     protected function _createSearchPostCodeTemplates($postcode)
     {
         $len = Mage::helper('tax')->getPostCodeSubStringLength();
-        $strlen = strlen($postcode);
+        $strlen = mb_strlen($postcode);
         if ($strlen > $len) {
-            $postcode = substr($postcode, 0, $len);
+            $postcode = mb_substr($postcode, 0, $len);
             $strlen = $len;
         }
 
         $strArr = array((string)$postcode, $postcode . '*');
         if ($strlen > 1) {
             for ($i = 1; $i < $strlen; $i++) {
-                $strArr[] = sprintf('%s*', substr($postcode, 0, - $i));
+                $strArr[] = sprintf('%s*', mb_substr($postcode, 0, - $i));
```

Comment:
can `mb_substr` take negative 3rd param? I don't see this described at https://www.php.net/manual/en/function.mb-substr.php

Reply:
I tried this:
```php
var_dump(substr("abcdef", 0, -1));  // = "abcde"
var_dump(substr("abcdef", 2, -1));  // = "cde"
var_dump(substr("abcdef", 4, -4));  // = false
var_dump(substr("abcdef", -3, -1)); // = "de"
echo "mb\n";
var_dump(mb_substr("abcdef", 0, -1));
var_dump(mb_substr("abcdef", 2, -1));
var_dump(mb_substr("abcdef", 4, -4));
var_dump(mb_substr("abcdef", -3, -1));
```

Result is:
```
$ php7.2 test.php
string(5) "abcde"
string(3) "cde"
bool(false)
string(2) "de"
mb
string(5) "abcde"
string(3) "cde"
string(0) ""
string(2) "de"

$ php7.3 test.php
string(5) "abcde"
string(3) "cde"
bool(false)
string(2) "de"
mb
string(5) "abcde"
string(3) "cde"
string(0) ""
string(2) "de"

$ php7.4 test.php
string(5) "abcde"
string(3) "cde"
bool(false)
string(2) "de"
mb
string(5) "abcde"
string(3) "cde"
string(0) ""
string(2) "de"

$ php8.0 test.php
string(5) "abcde"
string(3) "cde"
string(0) ""
string(2) "de"
mb
string(5) "abcde"
string(3) "cde"
string(0) ""
string(2) "de"
```
So, yes?